tests/project/test_config.py

```diff
@@ -145,6 +148,31 @@ def test_fields_are_parsed_correctly(project_definition_files, snapshot):
 )
 def test_schema_is_validated_for_version(data):
     with pytest.raises(SchemaValidationError) as err:
-        ProjectDefinition(**data)
+        build_project_definition(**data)
 
     assert "is not supported in given version" in str(err.value)
+
+
+def test_project_definition_v2_is_disabled():
+    assert FeatureFlag.ENABLE_PROJECT_DEFINITION_V2.is_enabled() == False
+    with pytest.raises(SchemaValidationError) as err:
+        build_project_definition(**{"definition_version": "2", "entities": {}})
+    assert "Version 2 is not supported" in str(err.value)
+
+
+def mock_config_key(key, value):
+    def get_config_side_effect_func(*args, **kwargs):
+        if kwargs["key"] == key:
+            return value
+        return kwargs["default"]
```

Comment:
for requests other than the mocked key, shouldn't we dispatch to the original `get_config_value` function instead of returning the default?

Reply:
Done. Also moved this code to the test utils folder.